src/Core/Core/Events/EventBus.cs

```diff
@@ -23,17 +24,35 @@ IExternalEventProducer externalEventProducer
         public Task Publish<TEvent>(TEvent @event, CancellationToken cancellationToken = default)
             where TEvent: IEvent
         {
-            return Publish(cancellationToken, @event);
+            return PublishInteral(@event, cancellationToken);
         }
 
         public async Task Publish(CancellationToken cancellationToken, params IEvent[] events)
         {
             foreach (var @event in events)
             {
-                await mediator.Publish(@event, cancellationToken);
+                await PublishInteral(@event, cancellationToken);
+            }
+        }
+
+        public async Task PublishParallel(CancellationToken cancellationToken, params IEvent[] events)
+        {
+            var tasks = new List<Task>();
+            foreach (var @event in events)
+            {
+                tasks.Add(Task.Run(() => PublishInteral(@event, cancellationToken)));
```

Comment:
@ledjon-behluli I think that Task.Run wrapper is redundant as PublishInternal already returns Task. But I'll sort it after the merge